### PR TITLE
feat: dockerize repository and release package on ghcr.io

### LINKED ISSUE #11: Feature: auto-detect docker builds
### Type of feature

Feature

### Current behavior

Users can enable `DISABLE_DOCKER` environment variable to disable docker builds.

### Desired solution

If a `Dockerfile` or any of the `docker login` variables do not exist, automatically skip pushing docker tags.

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #18: Feature: dockerize repository
### Type of feature

Feature

### Current behavior

you need `npm>6` installed in order to use this package

### Desired solution

dockerize this so it can run from a local directory without `node` dependencies

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #24: Feature: man in the middle buggy setups
### Type of feature

Refactor

### Current behavior

some plugins don't work as expected

### Desired solution

middleware all that before publishing with semantic-release

### Additional context

explore https://github.com/pmowrer/semantic-release-plugin-decorators

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by 0-vortex**
## What type of PR is this? (check all applicable)

- [ ] ♻️ Refactor
- [x] ✨ Feature
- [x] 🐛 Bug Fix
- [x] 👷 Optimization
- [x] 📝 Documentation Update
- [x] 🔖 Release
- [ ] 🚩 Other

## Description

This PR: 
- adds the ability to run our `semantic-release` configuration from a docker container
- adds the ability to run our `semantic-release` configuration from a GitHub action
- enables usage of our release tooling and configuration without installing any more dependencies
- simplifies the release workflow by removing almost all environment variables
- configuration is now aware of existing `Dockerfile` and `action.yml` configuration
- fixes the [skip ci] bug in `alpha` and `beta` pre-release branches
- fixes the npm ci build-cache using Bahmutov action
- adds better usage documentation and examples
- adds license update function and git asset configuration
- adds better changelog generation

## Related Tickets & Documents

closes #11 
closes #18
closes #24  

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [x] 📜 readme
- [ ] 📜 contributing.md
- [x] 📓 docs
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

## [optional] Are there any post-deployment tasks we need to perform?

- update maintainer docs
- write a dev.to blog post
- convince @mtfoley this action was designed for them

### CODE REVIEW

**Comment by 0-vortex** [comment]
@bdougie @mtfoley mind me pinging but you really need to see this, working on the docs right now but there are a lot of gaps due to rapid development, could really use your input!

**Comment by mtfoley** [comment]
So I read that in the original issue #18 that the purpose behind doing this is creating docker images so contributors don't need to deal with dependencies directly.  How would this change the workflow of cases where a contributor is adding dependencies to the project for features?

**Comment by 0-vortex** [comment]
> So I read that in the original issue #18 that the purpose behind doing this is creating docker images so contributors don't need to deal with dependencies directly. How would this change the workflow of cases where a contributor is adding dependencies to the project for features?

Other workflows would handle this - for now what this means is that with the simple squash flow enabled by your action we can simplify almost all releases with this workflow:

```yaml
name: "Release"

on:
  push:
    branches:
      - main

jobs:
  release:
    runs-on: ubuntu-latest
    steps:
      - name: "☁️ checkout repository"
        uses: actions/checkout@v2
        with:
          fetch-depth: 0

      - name: "🚀 release"
        id: semantic-release
        uses: docker://ghcr.io/open-sauced/semantic-release-conventional-config:3.0.0
        env:
          DISABLE_DOCKER: true
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          GIT_AUTHOR_NAME: ${{ github.event.commits[0].author.username }}
          GIT_AUTHOR_EMAIL: ${{ github.event.commits[0].author.email }}

      - name: 'test'
        run: |
          echo $SEMVER_VERSION
```

No more local dependencies, commit convention handled by your action, BOOM 🍕

**Comment by 0-vortex** [comment]
The bug fixes and general `semantic-release` quirks can be seen in [open-sauced-semantic-config-test](https://github.com/0-vortex/open-sauced-semantic-config-test/releases/tag/v3.19.0) and the performance improvements from running the release process in [semantic-release-docker-test](https://github.com/0-vortex/semantic-release-docker-test/actions/runs/1511111939) - both repositories I have created to be less spammy on open-sauced.

We really need `alpha` and `beta` branches enabled and documented, feel like they are so much easier to work with in the long run :D

File: .github/workflows/marketplace.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: "Action major version tag"
+
+on:
+  release:
+    types:
+      - published
```
Comment (0-vortex) [inline .github/workflows/marketplace.yml:6]: @bdougie we can get this workflow to trigger by changing the release `github.token` to admin level PAT but if the marketplace listing is unable to happen from semantic-release it's worth close to nothing to have half a workaround :D

**Comment by 0-vortex** [comment]
Linking this back to [discussion#1319](https://github.com/open-sauced/open-sauced/discussions/1319), is it worth doing this now, re-releasing at a later stage?

PS: It's ready for release except the pre-release branches!

File: .github/workflows/marketplace.yml (hunk 1 above)
Comment (0-vortex) [reply .github/workflows/marketplace.yml:6]: a better workaround is using `docker://` socket with ghcr.io, added those bits to the readme now

**Pull request merged**